### DISCUSSION TITLE
Is there a way to determine how much CPU, disk and network is being used by Vector?

### DISCUSSION THREAD

**Discussion by mkwan-amzn** [vectordotdev/vector#21346] [Q&A] [answered]
Hi Everyone, wondering if it is possible to figure out how much CPU, disk and network is being used specifically by Vector, and whether or not this can be broken up by Vector component. I did some testing with host metrics source but couldn't find this, although I may have missed it. 

I know the allocation tracing features lets you see memory allocation at the component level, and ultimately I'm wondering if something similar exists for the other metrics I mentioned. 

Thanks in advance for the time!

**Comment by jszwedko** [comment, marked as answer]
Unfortunately, I don't think Vector has much to offer, currently, for monitoring its own usage:

- Memory can somewhat measured by enabling allocation tracking, but I think there are places where Vector allocates memory that would not be covered
- I don't think there is anything for CPU, nor how we would even go about accomplishing that per-component aside from CPU profiling
- For network, we had discussed adding a metric for this in https://github.com/vectordotdev/vector/issues/11803 but this is yet to be implemented

Apologies that I can't be of more help!
